### ISSUE TITLE
Socket connection leak

### ISSUE THREAD

**Issue by marcoleong** [Microsoft/Ethr#4]
### Environment

Ethr version v0.1

- Linux 4.14.81-coreos Container Linux by CoreOS 1911.4.0 Rhyolite Container Linux by CoreOS 1911.4.0 (Rhyolite) 
- Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz GenuineIntel GNU/Linux

### What happend ?

Network performance degraded after socket reached CPU core limit. User have to restart server process to have normal performance data.

The connections will stuck at `CLOSE_WAIT` state after each test ran. This will reduce almost half the bandwidth, which in our case, bandwidth have gone done from `8 GBit/s` to `5 GBit/s`

Output of `netstat -noa`
```
tcp6	   0	  0 :::9999                 :::*                    LISTEN	117950/./ethr
tcp6  2861521	   0 10.33.3.25:9999         10.33.3.23:53702        CLOSE_WAIT  117950/./ethr
tcp6  2528001      0 10.33.3.25:9999         10.33.3.23:53762        CLOSE_WAIT  117950/./ethr
tcp6  2846801	   0 10.33.3.25:9999         10.33.3.23:53644        CLOSE_WAIT  117950/./ethr
```

### How to reproduce

1. Start server with `ethr -s`
2. Watch network connection with `watch 'netstat -noa | grep 9999'`
3. Start client connection with `ethr -c <ip_address>`, and repeat this multiple times, you should see the result after it reaches CPU core count on the server machine.

**Comment by pgmsft** [comment]
Thank you for reporting the issue. I would take a look at it tomorrow.

**Comment by pgmsft** [comment]
Thank you again for reporting the issue. This is fixed by the following commit: 6ea7c76e3b4b9562e2b25b4a0c8f5b1663148a28
